python_benchmarks/synthetic_benchmark/synthetic_benchmark.py

```diff
@@ -0,0 +1,488 @@
+class SyntheticObject:
+    def __init__(self):
+        self.image = None
+        self.fields = None
+        self.history = []
+
+    def transform_object(self, transform_object):
+        return self
+
+    def show(self, wait_key=0):
+        pass
+
+
+class BackGroundObject(SyntheticObject):
+    def __init__(self, *, num_rows, num_cols):
+        self.image = np.zeros((num_rows, num_cols), dtype=np.uint8)
+
+    def show(self, wait_key=0):
+        cv.imshow("BackGroundObject", self.image)
+        cv.waitKey(wait_key)
+
+
+class SyntheticCharuco(SyntheticObject):
+    def __init__(self, *, board_size, cell_img_size, square_marker_length_rate=0.5, dict_id=0):
+        self.board_size = board_size
+        self.square_marker_length_rate = square_marker_length_rate
+        self.dict_id = dict_id
+        self.dict = cv.aruco.getPredefinedDictionary(dict_id)
+        self.charuco_board = cv.aruco.CharucoBoard(board_size, 1., 1. * square_marker_length_rate, self.dict)
+        board_image_size = [board_size[0] * cell_img_size, board_size[1] * cell_img_size]
+        self.image = self.charuco_board.generateImage(board_image_size)
+        self.aruco_corners = (np.array(self.charuco_board.getObjPoints(), dtype=np.float32)
+                              * cell_img_size).reshape(-1, 3)[:, :-1]
+        self.aruco_ids = np.array(self.charuco_board.getIds())
+        self.chessboard_corners = (np.array(self.charuco_board.getChessboardCorners(), dtype=np.float32)
+                                   * cell_img_size)[:, :-1]
+        self.fields = {"board_size": None, "square_marker_length_rate": None, "dict_id": None, "aruco_corners": None,
+                       "aruco_ids": None, "chessboard_corners": None}
+        self.history = []
+
+    def transform_object(self, transform_object):
+        self.image = transform_object.transform_image(self.image)
+        self.aruco_corners = np.array(transform_object.transform_points(self.aruco_corners), dtype=np.float32)
+        self.chessboard_corners = np.array(transform_object.transform_points(self.chessboard_corners), dtype=np.float32)
+        if transform_object.name != "":
+            self.history.append(transform_object.name)
+        return self
+
+    def show(self, wait_key=0):
+        assert self.image is not None
+        image = np.copy(self.image)
+        aruco = np.array(self.aruco_corners.reshape(-1, 1, 4, 2), dtype=np.float32)
+        aruco = [el for el in aruco]
+
+        cv.aruco.drawDetectedMarkers(image, aruco)
+        chessboard_corners = self.chessboard_corners.reshape(-1, 1, 2)
+        cv.aruco.drawDetectedCornersCharuco(image, chessboard_corners)
+        cv.imshow("SyntheticCharuco", image)
+        cv.waitKey(wait_key)
+
+    def write(self, path="test", filename="test"):
+        for name in self.fields:
+            self.fields[name] = getattr(self, name)
+        with open(path+"/"+filename+'.txt', 'w') as fp:
+            json.dump(self.fields, fp, cls=NumpyEncoder)
```

Comment:
I propose to dump array of boards to be able to generate and check several boards on the same image

Reply:
added